src/test/sql/sql_pipeline_test.cpp

```diff
@@ -0,0 +1,470 @@
+ protected:
+  void SetUp() override {
+    _table_a = load_table("src/test/tables/int_float.tbl", 2);
+    StorageManager::get().add_table("table_a", _table_a);
+
+    _table_b = load_table("src/test/tables/int_float2.tbl", 2);
+    StorageManager::get().add_table("table_b", _table_b);
+
+    _join_result = std::make_shared<Table>();
+    _join_result->add_column("a", DataType::Int);
+    _join_result->add_column("b", DataType::Float);
+    _join_result->add_column("bb", DataType::Float);
+    _join_result->append({12345, 458.7f, 456.7f});
+    _join_result->append({12345, 458.7f, 457.7f});
+  }
+
+  std::shared_ptr<Table> _table_a;
+  std::shared_ptr<Table> _table_b;
+  std::shared_ptr<Table> _join_result;
+
+  const std::string _select_query_a = "SELECT * FROM table_a";
+  const std::string _invalid_sql = "SELECT FROM table_a";
+  const std::string _join_query =
+      "SELECT table_a.a, table_a.b, table_b.b AS bb FROM table_a, table_b WHERE table_a.a = table_b.a AND table_a.a "
+      "> 1000";
+
+  const std::vector<std::string> _column_names{"a", "b"};
+  const std::vector<std::string> _join_column_names{"a", "b", "bb"};
+
+  bool _subtree_is_equal(const std::shared_ptr<AbstractLQPNode>& got,
+                         const std::shared_ptr<AbstractLQPNode>& expected) {
+    if (got == nullptr && expected == nullptr) return true;
+    if (got == nullptr) return false;
+    if (expected == nullptr) return false;
+
+    if (got->type() != expected->type()) return false;
+    return _subtree_is_equal(got->left_child(), expected->left_child()) &&
+           _subtree_is_equal(got->right_child(), expected->right_child());
+  }
+
+  template <typename Functor>
+  bool _find_in_lqp(const std::shared_ptr<AbstractLQPNode>& node, Functor find_fn) {
+    if (node == nullptr) return false;
+    if (find_fn(node)) return true;
+    return _find_in_lqp(node->left_child(), find_fn) || _find_in_lqp(node->right_child(), find_fn);
+  }
+
+  template <typename Functor>
+  bool _find_in_query_plan(const std::shared_ptr<const AbstractOperator>& node, Functor find_fn) {
+    if (node == nullptr) return false;
+    if (find_fn(node)) return true;
+    return _find_in_query_plan(node->input_left(), find_fn) || _find_in_query_plan(node->input_right(), find_fn);
+  }
+
+  bool _contains_validate(const std::vector<std::shared_ptr<OperatorTask>>& tasks) {
+    for (const auto& task : tasks) {
+      if (std::dynamic_pointer_cast<Validate>(task->get_operator())) return true;
+    }
+    return false;
+  }
+
+  // This function is a slightly hacky way to check whether an LQP was optimized. This relies on JoinDetectionRule and
+  // could break if something is changed within the optimizer.
+  // It assumes that for the query: SELECT * from a, b WHERE a.a = b.a will be translated to a Cross Join with a filter
+  // predicate and then optimized to a Join.
+  std::function<bool(const std::shared_ptr<AbstractLQPNode>&)> _contains_cross =
```

Comment:
Could be a static function, or one in a anonymous namespace as well, no need to create an object here.